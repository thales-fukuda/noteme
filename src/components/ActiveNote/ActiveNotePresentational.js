import React, { Component } from 'react';
import PropTypes from 'prop-types';

import NoteWrapper from '../blocks/NoteWrapper';

class ActiveNotePresentational extends Component {
  constructor(props) {
    super(props);
    this.textTitle = React.createRef();
    this.textBody = React.createRef();
  }

  componentDidMount = () => {
    this.changeTextAreaSize(this.textTitle.current);
    this.changeTextAreaSize(this.textBody.current);
  }

  changeTextAreaSize = (el) => {
    const element = el;
    element.style.height = `${element.scrollHeight}px`;
  };

  handleChangeTitle = (e, id) => {
    e.stopPropagation();
    const { changeNoteTitle } = this.props;
    changeNoteTitle(id, e.target.value);
    this.changeTextAreaSize(e.target);
  };

  handleChangeBody = (e, id) => {
    e.stopPropagation();
    const { changeNoteBody } = this.props;
    changeNoteBody(id, e.target.value);
    this.changeTextAreaSize(e.target);
  };

  render() {
    const { notes, activeNote } = this.props;
    return (
      <NoteWrapper>
        <NoteWrapper.TitleText
          onChange={e => this.handleChangeTitle(e, notes[activeNote].id)}
          ref={this.textTitle}
          value={notes[activeNote].noteTitle}
        />
        <NoteWrapper.Text
          onChange={e => this.handleChangeBody(e, notes[activeNote].id)}
          ref={this.textBody}
          value={notes[activeNote].noteBody}
        >
        </NoteWrapper.Text>
      </NoteWrapper>
    );
  }
}

ActiveNotePresentational.propTypes = {
  notes: PropTypes.instanceOf(Array).isRequired,
  activeNote: PropTypes.number.isRequired,
  changeNoteTitle: PropTypes.func.isRequired,
  changeNoteBody: PropTypes.func.isRequired,
};

export default ActiveNotePresentational;
