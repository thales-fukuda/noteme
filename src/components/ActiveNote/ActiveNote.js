import { connect } from 'react-redux';
import { changeNoteTitle, changeNoteBody } from '../../redux/actions';
import ActiveNotePresentational from './ActiveNotePresentational';

const mapStateToProps = state => ({
  notes: state.notes.notes,
  activeNote: state.notes.activeNote,
});

const mapDispatchToProps = dispatch => ({
  changeNoteTitle: (id, title) => dispatch(changeNoteTitle(id, title)),
  changeNoteBody: (id, body) => dispatch(changeNoteBody(id, body)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ActiveNotePresentational);
