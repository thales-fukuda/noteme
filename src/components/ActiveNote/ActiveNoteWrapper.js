import { connect } from 'react-redux';
import { changeNoteTitle, changeNoteBody } from '../../redux/actions';
import ActiveNote from './ActiveNote';

const mapStateToProps = state => ({
  notes: state.notes.notes,
  activeNote: state.notes.activeNote,
});

const mapDispatchToProps = dispatch => ({
  changeNoteTitle: (id, title) => dispatch(changeNoteTitle(id, title)),
  changeNoteBody: (id, body) => dispatch(changeNoteBody(id, body)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ActiveNote);
