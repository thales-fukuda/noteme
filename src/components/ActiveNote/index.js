import ActiveNoteWrapper from './ActiveNoteWrapper';

export default ActiveNoteWrapper;
