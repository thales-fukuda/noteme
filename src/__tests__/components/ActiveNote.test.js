/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ActiveNote from '../../components/ActiveNote/ActiveNote';

describe('<ActiveNote/>', () => {
  describe('render()', () => {
    it('renders the component', () => {
      const fakeProps = {
        notes: [],
        activeNote: null,
        changeNoteTitle: () => {},
        changeNoteBody: () => {},
      };
      const wrapper = shallow(<ActiveNote {...fakeProps} />).dive();
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
