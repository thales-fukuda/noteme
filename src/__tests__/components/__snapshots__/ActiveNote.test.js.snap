// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActiveNote/> render() renders the component 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "Text": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
  min-height: calc(100vh - 190px);
  margin-top: 30px;
  background: transparent;
  font-family: 'Slabo 27px';
  font-size: 24px;
  width: 100%;
  height: auto;
",
          ],
        },
        "displayName": "styled.textarea",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "textarea",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "TitleText": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  height: 40px;
  width: 100%;
  background-color: transparent;
  border: none;
  font-family: 'Lato';
  font-size: 32px;
  font-weight: bold;
",
          ],
        },
        "displayName": "styled.input",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "input",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": true,
        "rules": Array [
          "
  margin-top: 70px;
  padding: 35px;
  box-sizing: border-box;
  overflow-x: hidden;

  @media ( min-width: 726px) {
    margin-left: 40%;
    margin-top: 0;
  }
",
        ],
      },
      "displayName": "styled.article",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "article",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
